src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeonardoModel.scala

```diff
@@ -276,9 +274,70 @@ object LeonardoJsonSupport extends SprayJsonSupport with DefaultJsonProtocol {
 
   implicit val ClusterErrorFormat = jsonFormat3(ClusterError.apply)
 
-  implicit val ClusterFormat = jsonFormat22(Cluster.apply)
+//  implicit val DataprocInfoFormat = jsonFormat3(DataprocInfo.apply)
+//
+//  implicit val AuditInfoFormat = jsonFormat4(AuditInfo.apply)
+//
+//  implicit val ClusterFormat = jsonFormat17(Cluster.apply)
 
   implicit val DefaultLabelsFormat = jsonFormat6(DefaultLabels.apply)
 
   implicit val ClusterInitValuesFormat = jsonFormat20(ClusterInitValues.apply)
+
+  implicit object Foo extends RootJsonFormat[Cluster] {
+    override def read(json: JsValue): Cluster = {
+      json match {
+        case JsObject(fields: Map[String, JsValue]) =>
+          Cluster(
+            fields("id").convertTo[Long],
+            fields("clusterName").convertTo[ClusterName],
+            fields("googleProject").convertTo[GoogleProject],
+            fields("serviceAccountInfo").convertTo[ServiceAccountInfo],
+            DataprocInfo(fields("googleId").convertTo[Option[UUID]],
+                          fields("operationName").convertTo[Option[OperationName]],
+                          fields("stagingBucket").convertTo[Option[GcsBucketName]],
+                          fields("hostIp").convertTo[Option[IP]]),
+            AuditInfo(fields("creator").convertTo[WorkbenchEmail],
+                      fields("createdDate").convertTo[Instant],
+                      fields("destroyedDate").convertTo[Option[Instant]],
+                      fields("dateAccessed").convertTo[Instant]),
+            fields("machineConfig").convertTo[MachineConfig],
+            fields("clusterUrl").convertTo[URL],
+            fields("status").convertTo[ClusterStatus],
+            fields("labels").convertTo[LabelMap],
+            fields("jupyterExtensionUri").convertTo[Option[GcsPath]],
+            fields("jupyterUserScriptUri").convertTo[Option[GcsPath]],
+            fields("errors").convertTo[List[ClusterError]],
+            fields("instances").convertTo[Set[Instance]],
+            fields("userJupyterExtensionConfig").convertTo[Option[UserJupyterExtensionConfig]],
+            fields("autopauseThreshold").convertTo[Int])
+        case _ => deserializationError("Cluster expected")
```

Comment:
I couldn't be sure if the error message would be sufficiently explanatory in the failure context. It looks like we classify any input but a map of strings to values as `deserializationError` so perhaps we can say something like
> Expected Cluster fields as a String->Value map

Reply:
"Cluster expected as a JsObject" sounds like an reasonable error message here